From: Werner Koch <wk@gnupg.org>
Date: Mon, 13 Jun 2011 12:35:30 +0000 (+0200)
Subject: Fix a for a bug fix in the latest Libgcrypt.
X-Git-Tag: gnupg-2.0.18~13
X-Git-Url: http://git.gnupg.org/cgi-bin/gitweb.cgi?p=gnupg.git;a=commitdiff_plain;h=13290b0e0fcf3a493e4848b29329d56b69bc4dd9

Fix a for a bug fix in the latest Libgcrypt.

        * pkglue.c (mpi_from_sexp, pk_decrypt): Use GCRYMPI_FMT_USG for
	gcry_sexp_nth_mpi.  This fixes a problem with a recent bug fix in
	Libgcrypt.
---
 
diff --git a/g10/pkglue.c b/g10/pkglue.c
index cbfe21e..5c47511 100644
--- a/g10/pkglue.c
+++ b/g10/pkglue.c
@@ -34,10 +34,10 @@ mpi_from_sexp (gcry_sexp_t sexp, const char * item)
 {
   gcry_sexp_t list;
   gcry_mpi_t data;
-  
+
   list = gcry_sexp_find_token (sexp, item, 0);
   assert (list);
-  data = gcry_sexp_nth_mpi (list, 1, 0);
+  data = gcry_sexp_nth_mpi (list, 1, GCRYMPI_FMT_USG);
   assert (data);
   gcry_sexp_release (list);
   return data;
@@ -293,7 +293,7 @@ pk_decrypt (int algo, gcry_mpi_t * result, gcry_mpi_t * data,
   if (rc)
     return rc;
 
-  *result = gcry_sexp_nth_mpi (s_plain, 0, 0);
+  *result = gcry_sexp_nth_mpi (s_plain, 0, GCRYMPI_FMT_USG);
   gcry_sexp_release (s_plain);
   if (!*result)
     return -1;			/* oops */
